src/core/diffusion_grid.h

```diff
@@ -108,6 +111,13 @@ class DiffusionGrid {
           "] will result in unphysical behavior (diffusion coefficient = ",
           (1 - dc_[0]), ", resolution = ", resolution_,
           "). Please refer to the user guide for more information.");
+    } else if ( (int)floor(diffusion_step_) != (int)diffusion_step_ || diffusion_step_ <= (double)0.0 || diffusion_step_ > (double)10.0) {
```

Comment:
The `(double)` casts are not needed.
What are you trying to achieve with `(int)floor(diffusion_step_) != (int)diffusion_step_`

Reply:
The idea behind this was to ensure that someone is not using a floating point for the diffusion_steps in Runge-Kutta, but I realise now it would be better to use a checker that converts it to a string then searches for a decimal. I believe for now this can be removed.

Reply:
If `diffusion_step_` must not be a decimal why do you define it as double?
Change it to an int or unsigned int